### ISSUE TITLE
Develop tools to query systematic maps.

### ISSUE THREAD

**Issue by fjaviersanchez** [LSSTDESC/DC2_Repo#26]
@slosar @damonge (and probably @joezuntz @mdschneider) would want to have capabilities to query different observational effects (sky-brightness in the different bands, seeing, bright stars, dust, etc) as 2D maps in the coadds. Right now, this information is at the per-visit/sensor level but we would need to propagate these into the coadds/HEALPix (or other scheme?)

For DC1 I was generating maps following the recipe:

- Check the value of the quantity of interest at the (ra,dec) of the center of the sensor and put this into its corresponding healpix pixel.

- Compute average and std in each pixel.

The caveat of this approach is that the geometry of each sensor is not properly propagated (as well as the overlaps). Is this enough for DC2? Do we want to develop something more complicated?

**Comment by slosar** [comment]
This is supposed to be @erykoff territory. He has been in touch with @TallJimbo regarding this.

**Comment by katrinheitmann** [comment]
@fjaviersanchez Do you know what happened to this? Is what has been developed so far sufficient? If not, we need to work on this quickly :-)

**Comment by fjaviersanchez** [comment]
@katrinheitmann, thanks for the reminder: As a summary, I think that we have the tools to deliver this but I don't know who would be responsible nor the preferred format.

We have software to do this in the HSC-LSS repo. I also have a new script that propagates the geometry correctly to `flatskymaps` as defined in the aforementioned repo that I have been using for DC1 data (which I should put in a repo). Both programs are quite slow (although a lot of processes can be run in parallel so that shortens the wait) since you have to loop over all single-visits (either by reading their WCS or by generating them).

Having said so, should we produce these maps when run 2.1i is available (I have a version that can produce them before the images are produced and uses the sims stack to produce the WCS but I trust more the version that reads the WCS from the calexps so there are no mismatches between the coordinates of the measured objects and the observing conditions) or is this something that each working group will produce, or is this something that DM will produce? Should we produce the maps for run1.x?

In addition, @erykoff and I have been developing a sparse implementation of HEALPix in python [here](https://github.com/LSSTDESC/healsparse) but we haven't tried to run this in DC2 data to produce maps yet. If this is something in which the Science WGs are interested in I'd be happy to run them (I would have to make some changes in the code that I use for DC1 though).

@rmandelb, @slosar, @damonge, @chihway, @timeifler, questions for you (and apologies if I miss any other interested parties): Would you be interested in having these "systematic" maps available for run 1.x (< 30 sq-deg)? If so, would you like them in flat-sky with ~1-2 arcmin resolution? Would this format be useful for run 2.1i (~300 sq-deg)? (I suspect that we'd want something different than flat-sky for this run but I thought it'd be worth asking). What would be the timescale in which it would be useful to have such maps?

Finally, I am not sure if this issue belongs more to analysis rather than production. Please feel free to close it and we can open a follow-up issue in the DC2-analysis repo if you consider that this doesn't belong here.
